src/main/java/cms/gov/madie/measure/resources/MeasureVersionController.java

```diff
@@ -32,13 +34,17 @@ public ResponseEntity<Measure> createVersion(
   }
 
   @GetMapping("/{id}/version")
-  public ResponseEntity<Measure> checkValidVersion(
+  public ResponseEntity<Void> checkValidVersion(
       @PathVariable("id") String id,
       @RequestParam String versionType,
       Principal principal,
-      @RequestHeader("Authorization") String accessToken)
-      throws Exception {
-    return versionService.checkValidVersioning(id, versionType, principal.getName(), accessToken);
+      @RequestHeader("Authorization") String accessToken) {
+    var validationResult =
+        versionService.checkValidVersioning(id, versionType, principal.getName(), accessToken);
```

Comment:
Wondering why we can't verify if measure is eligible for Versioning in the front end ?

Reply:
It probably could. Depends on our approach to APIs.

Reply:
Ofcourse yeah. No hard opinion here, but just wanted to ask because this is introducing some latency.

Reply:
Yeah, I didn't think of this. We already support users to share measures, So it is a possible scenario, and a validation check up front using API does reduce the window for possible concurrency issues.

Never worked on it, but love to implement some sort of DB triggers and events to handle the object changes dynamically on the UI. I guess that is a discussion for another day.